lib/macho/exceptions.rb

```diff
@@ -3,6 +3,10 @@ module MachO
   class MachOError < RuntimeError
   end
 
+  # A generic Mach-O error during file modification.
+  class MachOModificationError < MachOError
```

Comment:
> We could also consider a name different from MachOModificationError that better reflects the nature of these exceptions.

This might get us uncomfortably close to checked and unchecked exceptions a la Java, but how about `MachONonFatalModificationError` (and `MachOFatalModificationError` for things like `HeaderPadError`)?

Agreed on the documentation :+1: